src/app/components/educational-programs/create-educational-program/create-educational-program.component.ts

```diff
@@ -0,0 +1,74 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, FormArray } from '@angular/forms';
+import { Subscription } from 'rxjs';
+import { pairwise, startWith } from 'rxjs/operators';
+import { Validators } from '@angular/forms';
+
+@Component({
+  selector: 'app-create-educational-program',
+  templateUrl: './create-educational-program.component.html',
+  styleUrls: ['./create-educational-program.component.scss'],
+})
+export class CreateEducationalProgramComponent implements OnInit {
+  programForm: FormGroup = this.fb.group({
+    acceptancePeriodStartDate: ['', [Validators.required]],
+    acceptancePeriodEndDate: ['', [Validators.required]],
+    programsPeriodStartDate: ['', [Validators.required]],
+    programsPeriodEndDate: ['', [Validators.required]],
+    name: ['', [Validators.required]],
+    numberOfPositions: 1,
+    positions: this.fb.array([
+      this.fb.group({
+        positionName: '',
+        descriptionAndRequirements: '',
+      }),
+    ]),
+  });
+
+  private numberOfPositionsSubscription: Subscription = new Subscription();
+
+  constructor(private fb: FormBuilder) {}
+
+  ngOnInit(): void {
+    this.onNumberOfPositionsChanges(); // Used to add or remove positions/techonologies inputs
+  }
+
+  ngOnDestroy() {
+    this.numberOfPositionsSubscription.unsubscribe();
+  }
+
+  get positions() {
+    return this.programForm.get('positions') as FormArray;
+  }
+
+  addPosition() {
+    const position = this.fb.group({
+      positionName: ['', [Validators.required]],
+      descriptionAndRequirements: ['', [Validators.required]],
+    });
+
+    this.positions.push(position);
+  }
+
+  deletePosition(i: any) {
+    this.positions.removeAt(i);
+    console.log('detePosition, i = ', i);
```

Comment:
please do not use `console.log` in production code. If it is necessary for some reason, add comment about it

Reply:
My bad, forgot to delete it after testing